projects/vdk-control-cli/tests/vdk/internal/control/command_groups/job/test_download_job.py

```diff
@@ -70,6 +82,11 @@ def test_download_job_does_not_exist(httpserver: PluginHTTPServer, tmpdir: Local
         uri="/data-jobs/for-team/test-team/jobs/test-job/sources", method="GET"
     ).respond_with_response(Response(status=404))
 
+    # Mock a 404 response for the keytab endpoint
```

Comment:
write separate tests methods test_download_job_does_not_exist and test_download_key_does_not_exist

Otherwise how do you know which one is handled appropriately